pkg/pillar/zedUpload/azureutil/azure.go

```diff
@@ -34,6 +36,30 @@ type UpdateStats struct {
 // NotifChan is the uploading/downloading progress notification channel
 type NotifChan chan UpdateStats
 
+// ChunkStats contians the details of Chunks being downloaded
+type ChunkStats struct {
+	Size  int64  // complete size to upload/download
+	Chunk []byte // chunk data being uploaded/downloaded
+}
+
+// ChunkWriter encapsulates the Chunk writing parameters
+type ChunkWriter struct {
+	data      ChunkStats
+	prgNotify ChunkStatsChan
+}
+
+// ChunkStatsChan is the uploading/downloading progress notification channel
+type ChunkStatsChan chan ChunkStats
+
+// Write methods recieves the chunk data and passed it to the the caller via channel provided
+func (cw *ChunkWriter) Write(chunk []byte) (n int, err error) {
+	if cw.prgNotify != nil {
```

Comment:
Duplicate code is one issue; duplicate code with different parameter names is worse.
Could we dine the ChunkWriter and related definitions in some common code across the clouds?